### PR TITLE
Mrproper/harmonize declarations

### PR DESCRIPTION
**Pull request by serge-sans-paille**
This one has been rebased on the nary expressions PR, only the last commit is relevant

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
@pbrunet travis is OK, except for a PEP8 issue due to pep8 upgrade, fixed in #390

**Comment by pbrunet** [comment]
Can you rebase this one?
Also : https://github.com/serge-sans-paille/pythran/commit/69d28a1553f80534c7c17098efdaceaa2a0485de#commitcomment-9655977

File: pythran/pythonic/types/numpy_expr.hpp [hunk 1]
```diff
@@ -89,10 +89,8 @@ namespace pythonic {
 
                 template<int... I>
                 long _flat_size(utils::seq<I...>) const {
-                  std::array<long, sizeof...(I)> sizes = {
-                    std::get<I-1>(args).flat_size()...
-                  };
-                  return *std::max_element(sizes.begin(), sizes.end());
+                  long const sizes[] = {std::get<I-1>(args).flat_size()...};
+                  return *std::max_element(std::begin(sizes), std::end(sizes));
```
Comment (serge-sans-paille) [inline pythran/pythonic/types/numpy_expr.hpp]: *this* is the correct commit. So it's OK

**Comment by pbrunet** [comment]
It looks like an SIMD issue remains (missing load function for types::array)

**Comment by serge-sans-paille** [comment]
looks good to travis!

**Pull request merged**